### PR TITLE
Fix #15483: Alert form destination selection improvements

### LINKED ISSUE #15483: UI: Alert form destination selection improvements
**Affected module**

UI

**Describe the bug**

- [x] Change the `Generic` option to `Webhook`

Screenshot 2024-03-07 at 3 00 15 PM

- [ ] The selected option labels are not showing properly for the external destination options

Screenshot 2024-03-07 at 3 03 01 PM

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
Fixes #15483 

I worked on the following:

- [x] Fix the selected option labels not showing properly for the external destination options

Screenshot 2024-03-08 at 5 20 14 PM

- [x] Changed the `Generic` option label to `Webhook` for better understanding.

Screenshot 2024-03-08 at 5 21 17 PM

#
### Type of change:

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/AddObservabilityPage/DestinationFormItem/DestinationSelectItem/DestinationSelectItem.tsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      )}
+    </>
+  );
+
+  const isEditMode = useMemo(() => !isEmpty(fqn), [fqn]);
+
+  // Using recommended method here https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
+  // To execute code after mounting instead of using useRef
+  const measuredRef = useCallback((node) => {
+    if (node !== null) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/AddObservabilityPage/DestinationFormItem/DestinationSelectItem/DestinationSelectItem.tsx:150]: Not sure why we need this I checked the link from the comment and it's recommended to use this method for measuring DOM node 
But in the function I don't see we are measuring anything

Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/AddObservabilityPage/DestinationFormItem/DestinationSelectItem/DestinationSelectItem.tsx:150]: Sorry, the name is a bit misLeading.
So basically the code inside is supposed to be executed after the component is mounted. But in the mentioned link in the comment, its not recommended to use useRef for that logic, like we do elsewhere in the application (isMounted.current)..

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/pages/AddObservabilityPage/DestinationFormItem/DestinationSelectItem/DestinationSelectItem.tsx:150]: I would like to know more about this usecase. Not preferring to use above method either

File: openmetadata-ui/src/main/resources/ui/cypress/common/AlertUtils.ts [hunk 2]
```diff
@@ -370,7 +370,10 @@ export const verifyAlertDetails = (
       // Check Destination type
       cy.get(
         `[data-testid="destination-${destination.category}"] [data-testid="destination-type"]`
-      ).should('contain', startCase(destination.type));
+      ).should(
+        'contain',
+        destination.type === 'Generic' ? 'Webhook' : startCase(destination.type)
```
Comment (harshach) [inline openmetadata-ui/src/main/resources/ui/cypress/common/AlertUtils.ts:375]: if the backend is giving you "Generic" then we need to change it in the backend and not have conditions like these in the UI cc @mohityadav766

**Review by harshach — CHANGES_REQUESTED** [review]
Lets add the necessary changes in the backend

**Pull request merged**